### PR TITLE
HHH-14632 - Call statistics.queryPlanCacheHit and statistics.queryPlanCacheMiss for FilterQueryPlan and NativeSQLQueryPlan

### PR DESCRIPTION
**Pull request by vladmihalcea**
https://hibernate.atlassian.net/browse/HHH-14632

### CODE REVIEW

**Comment by vladmihalcea** [comment]
@beikov If you have time, can you take a look at this? I ran all tests and checks, and everything worked like a charm.

The advantage of having this merged is that the plan cache hit/miss stats will be more accurate.

**Comment by NathanQingyangXu** [comment]
> @beikov If you have time, can you take a look at this? I ran all tests and checks, and everything worked like a charm.
> 
> The advantage of having this merged is that the plan cache hit/miss stats will be more accurate.

There is a Mariadb testing failure for now.

**Comment by vladmihalcea** [comment]
@NathanQingyangXu I couldn't get the MariaDB error, and when running it locally on MySQL, it was all working just fine. 

I modified the native SQL queries and set the table name explicitly, maybe it will work better this time.

**Comment by vladmihalcea** [comment]
@NathanQingyangXu All good now!

File: hibernate-core/src/test/java/org/hibernate/stat/internal/QueryPlanCacheStatisticsTest.java [hunk 1]
```diff
@@ -171,6 +172,54 @@ public void testCreateQueryHitCount() {
...
+
+		doInJPA( this::entityManagerFactory, entityManager -> {
+
+			List<Employee> employees = entityManager.createNativeQuery(
+				"select * from employee e", Employee.class )
+			.getResultList();
+
+			assertEquals( 5, employees.size() );
+
+			//The miss count is still 1, as no we got the query plan from the cache
```
Comment (NathanQingyangXu) [inline hibernate-core/src/test/java/org/hibernate/stat/internal/QueryPlanCacheStatisticsTest.java:202]: as no -> as now?

File: hibernate-core/src/test/java/org/hibernate/stat/internal/QueryPlanCacheStatisticsTest.java [hunk 2]
```diff
@@ -171,6 +172,54 @@ public void testCreateQueryHitCount() {
...
```
Comment (NathanQingyangXu) [inline hibernate-core/src/test/java/org/hibernate/stat/internal/QueryPlanCacheStatisticsTest.java:216]: as now?

File: hibernate-core/src/main/java/org/hibernate/engine/query/spi/QueryPlanCache.java [hunk 3]
```diff
@@ -200,17 +200,26 @@ public FilterQueryPlan getFilterQueryPlan(
 			Map<String,Filter> enabledFilters) throws QueryException, MappingException {
 		final FilterQueryPlanKey key =  new FilterQueryPlanKey( filterString, collectionRole, shallow, enabledFilters );
 		FilterQueryPlan value = (FilterQueryPlan) queryPlanCache.get( key );
+		final StatisticsImplementor statistics = factory.getStatistics();
+		boolean stats = statistics.isStatisticsEnabled();
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/engine/query/spi/QueryPlanCache.java:204]: the 'stats' seems too terse and might be renamed to 'isStatEnabled' or stuff like that? `final` keyword is desirable as well.

File: hibernate-core/src/test/java/org/hibernate/stat/internal/QueryPlanCacheStatisticsTest.java [hunk 4]
```diff
@@ -171,6 +172,54 @@ public void testCreateQueryHitCount() {
...
+	public void testCreateNativeQueryHitCount() {
+		statistics.clear();
+
+		doInJPA( this::entityManagerFactory, entityManager -> {
+
+			List<Employee> employees = entityManager.createNativeQuery(
+				"select * from employee e", Employee.class )
+			.getResultList();
+
+			assertEquals( 5, employees.size() );
```
Comment (NathanQingyangXu) [inline hibernate-core/src/test/java/org/hibernate/stat/internal/QueryPlanCacheStatisticsTest.java:186]: Is the above assertion necessary? Seems irrelevant to cache stats assertion.

**Comment by NathanQingyangXu** [comment]
Overall looks great, except for some minor details. I would approve once they are attended to.

File: hibernate-core/src/test/java/org/hibernate/stat/internal/QueryPlanCacheStatisticsTest.java (hunk 1 above)
Comment (vladmihalcea) [reply hibernate-core/src/test/java/org/hibernate/stat/internal/QueryPlanCacheStatisticsTest.java:202]: Yes, we should fix those typos too

**Comment by vladmihalcea** [comment]
@NathanQingyangXu Fixed and rebased.

**Review by beikov — APPROVED** [review]
LGTM, thanks for the great work!

**Comment by vladmihalcea** [comment]
Thanks @beikov and @NathanQingyangXu
